### ISSUE TITLE
Changing error message format

### ISSUE THREAD

**Issue by Trevoke** [ruby-json-schema/json-schema#97]
It would be nice to have a way to customize the formatting of the error messages.

**Comment by haihappen** [comment]
Any progress on this? I would love the replace the validation logic in my rails-api app with JSON::Validator :smile:

**Comment by iainbeeston** [comment]
Haven't had a chance yet.

If you're interested I did make a gem that used Json Schema for active model validations:

https://github.com/iainbeeston/json_validator

**Comment by haihappen** [comment]
Wow, this is incredible useful! :)

**Comment by GrooveStomp** [comment]
i18n support would be great, but I'm not sure it's quite what my team is looking for.
What we'd like to see is the ability to call `#fully_validate` and get a list of error objects in return, like this:
```Ruby
[
  {
    fragment: '#/',
    property: 'city',
    failed_attribute: 'required'
  }
]
```
That way we could meaningfully construct our own keys for translation.
Or would your idealized implementation obviate this approach, @iainbeeston ?
(PS - Thanks for the gem!)

**Comment by iainbeeston** [comment]
I see. That's a good idea. At the minimum I can make sure errors as objects
includes the i18n keys for the error messages

**Comment by GrooveStomp** [comment]
@iainbeeston That sounds great.
I may forge ahead with a fork here.  Are you amenable to pull requests?

**Comment by iainbeeston** [comment]
Yes of course - go ahead! A minimal implementation would be to add the
fields you suggested to the errors object and later I can use the error
type as the key for an i18n translation

**Comment by vasilakisfil** [comment]
any progress on that ? I think it's important as more and more people are using JSON schema for their API validation :)

**Comment by GrooveStomp** [comment]
We've been using a fork for several months now that implements behavior based on the above.  I have some work left on it before submitting, but you can view my changes here: https://github.com/GrooveStomp/json-schema/pull/1

**Comment by vasilakisfil** [comment]
@GrooveStomp please do I think your work is vital to the community :)

**Comment by iainbeeston** [comment]
Thanks for sharing @GrooveStomp , I still haven't had time to work on my own solution, but my plan was to make all of the error messages customisable, as well as the property names.

I'd like to put some time into this in the near future but unfortunately there are performance issues with the gem that are even more pressing
